tests/basic-test.sh

```diff
@@ -459,6 +459,25 @@ assert_file_has_content checkout-test-union-add/union-add-test 'existing file fo
 assert_file_has_content checkout-test-union-add/union-add-test2 'another file for union add testing'
 echo "ok checkout union add"
 
+# Create some new files for testing
+cd ${test_tmpdir}
+mkdir add-dirs-test
+mkdir add-dirs-test/test_one
+chmod a+w add-dirs-test/test_one
+touch add-dirs-test/test_one/test_file
+$OSTREE commit ${COMMIT_ARGS} -b test-add-dirs --tree=dir=add-dirs-test
+$OSTREE checkout --add-dirs test-add-dirs checkout-test-add-dirs
+echo "ok adding new directories and new file"
+# Make a new file, and try the checkout again
+touch add-dirs-test/test_one/another_test_file
+$OSTREE commit ${COMMIT_ARGS} -b test-add-dirs --tree=dir=add-dirs-test
+# Check out the latest commit, should fail due to existing files
+if $OSTREE checkout --add-dirs test-add-dirs checkout-test-add-dirs 2> err.txt; then
+    assert_not_reached "checking out files unexpectedly succedded!"
+fi
+assert_file_has_content err.txt 'File exists'
```

Comment:
Let's also do a `--union` checkout here and assert that that works?

Also can you squash these two commits together?

Reply:
yup, changed based on the comment, but there are something that I noticed while adding the check for union.

First, when you do a touch for multiple files, it seems that the content between those files were shared, ( I did a test with `touch a` and `touch b`, then whenever I change a, b's content also get changed)

Because of that,  I removed touch and replaced it with echo( which I have seen it working fine so far) 

Second, when I do a modification on the checked out files, it seems like when you do union, the file content will not be overwritten because of the hard link. Therefore, I have to manually remove the file to break the hardlink. Is that wanted behavior?

If I have done something wrong in the code or some of my observations were not suppose to happen, feel free to point it out. :)

Reply:
Yeah, generally a good idea to use separate content.

As far as unioning....hmm, I'm not sure exactly what you mean there; does it only happen when the files are hardlinked?  There is this comment https://github.com/ostreedev/ostree/blob/355e8516b04801caf50b0b8a218fc8f6321d8d05/src/libostree/ostree-repo-checkout.c#L341
which I think bit me before.

Reply:
yea, I need to remove the hardlinked files and create the one with same name to make file overwrite work.
which is the code section below: 
```
# First 'break' the hard link for the previous files
rm checkout-test-add-dirs/test_one/test_file
echo 'file for testing union mode alongwith add-dirs mode' > checkout-test-add-dirs/test_one/test_file```